src/lib.rs

```diff
@@ -476,6 +476,52 @@ impl<T: Debug + PrimInt + One + Zero> Vob<T> {
         }
     }
 
+    /// Returns contents of Vob as Vec of bytes.
+    ///
+    /// For a Vob of length n, the first bit will be the 0th element,
+    /// the last bit would be the (n-1)th element.
+    ///
+    /// # Examples
+    /// ```
+    /// use vob::Vob;
+    /// let mut x = Vob::from_elem(5, false);
+    /// x.set(2, true);
+    /// x.set(4, true);
+    /// assert_eq!(x.to_bytes(), [0b00101000]);
+    /// ```
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut v: Vec<u8> = Vec::new();
+        for j in 0..self.vec.len() {
+            let current_block = self.vec[j];
+            let bytes_per_block = bytes_per_block::<T>();
+            for i in (0..bytes_per_block).rev() {
```

Comment:
Here's an idea: if we treat the block as an array of bytes we *might* be able to speed things up a bit. Here's a sketch of the idea https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=854448102030bc9658a7a069e6011a6b -- notice that we'd need to see benchmark numbers before and after to know if the extra complexity is worth it.